typescript/lib/config.ts

```diff
@@ -31,35 +31,47 @@ export type SaveOptions = {
   serializeOutputs?: boolean;
 };
 
-ModelParserRegistry.registerModelParser(new OpenAIModelParser(), [
-  "babbage-002",
-  "davinci-002",
-  "gpt-3.5-turbo-instruct",
-  "text-davinci-003",
-  "text-davinci-002",
-  "text-davinci-001",
-  "code-davinci-002",
-  "text-curie-001",
-  "text-babbage-001",
-  "text-ada-001",
-]);
-
-ModelParserRegistry.registerModelParser(new OpenAIChatModelParser(), [
-  "gpt-4",
-  "gpt-4-0314",
-  "gpt-4-0613",
-  "gpt-4-32k",
-  "gpt-4-32k-0314",
-  "gpt-4-32k-0613",
-  "gpt-3.5-turbo",
-  "gpt-3.5-turbo-16k",
-  "gpt-3.5-turbo-0301",
-  "gpt-3.5-turbo-0613",
-  "gpt-3.5-turbo-16k-0613",
-]);
-ModelParserRegistry.registerModelParser(new HuggingFaceTextGenerationParser());
-ModelParserRegistry.registerModelParser(new PaLMTextParser());
-
+export function registerDefaultModelParsers() {
+  ModelParserRegistry.registerModelParser(new OpenAIModelParser(), [
+    "babbage-002",
+    "davinci-002",
+    "gpt-3.5-turbo-instruct",
+    "text-davinci-003",
+    "text-davinci-002",
+    "text-davinci-001",
+    "code-davinci-002",
+    "text-curie-001",
+    "text-babbage-001",
+    "text-ada-001",
+  ]);
+
+  ModelParserRegistry.registerModelParser(new OpenAIChatModelParser(), [
+    "gpt-4",
+    "gpt-4-0314",
+    "gpt-4-0613",
+    "gpt-4-32k",
+    "gpt-4-32k-0314",
+    "gpt-4-32k-0613",
+    "gpt-3.5-turbo",
+    "gpt-3.5-turbo-16k",
+    "gpt-3.5-turbo-0301",
+    "gpt-3.5-turbo-0613",
+    "gpt-3.5-turbo-16k-0613",
+  ]);
+
+  ModelParserRegistry.registerModelParser(
+    new OpenAIChatModelParser({
+      apiKey: getAPIKeyFromEnv("ANYSCALE_ENDPOINT_API_KEY"),
+      baseURL: "https://api.endpoints.anyscale.com/v1",
+    }),
+    ["AnyscaleEndpoint"]
```

Comment:
The way the anyscale endpoint works is we pass the "model" to it. So the settings in the aiconfig would be something like:
```
{
      "name": "gen_itinerary",
      "input": "Generate an itinerary ordered by {{order_by}} for these activities: {{get_activities.output}}.",
      "metadata": {
        "model": {
          "name": "AnyscaleEndpoint",
          "settings": {
            "model": "mistralai/Mixtral-8x7B-Instruct-v0.1"
          }
        },
        "parameters": {
          "order_by": "geographic location"
        }
      }
    }
```

Alternatively, we create a `model_parsers` mapping in the aiconfig, and use those model IDs directly:
```
"model_parsers": {
      "meta-llama/Llama-2-7b-chat-hf": "AnyscaleEndpoint",
      "meta-llama/Llama-2-13b-chat-hf": "AnyscaleEndpoint",
      "Meta-Llama/Llama-Guard-7b": "AnyscaleEndpoint",
      "meta-llama/Llama-2-70b-chat-hf": "AnyscaleEndpoint",
      "Open-Orca/Mistral-7B-OpenOrca": "AnyscaleEndpoint",
      "codellama/CodeLlama-34b-Instruct-hf": "AnyscaleEndpoint",
      "HuggingFaceH4/zephyr-7b-beta": "AnyscaleEndpoint",
      "mistralai/Mistral-7B-Instruct-v0.1": "AnyscaleEndpoint",
      "mistralai/Mixtral-8x7B-Instruct-v0.1": "AnyscaleEndpoint",
      "thenlper/gte-large": "AnyscaleEndpoint"
    },
```

I think this model parser works similarly to a HF task -- i.e. it aggregates different models under a single model parser. We will need to think about how we handle this in the UI (ideally we do it the same way we will handle HF tasks).

I was thinking a ModelParser can have a `list_models` API which returns the list of model IDs that it supports. Then the `list_models` endpoint for the local editor will be able to enumerate those properly.

Reply:
I still think it's a bit strange how we're able to conflate model and model parser in the config itself -- this is another way we are able to represent the same concept in multiple different ways in the config & makes it pretty hard to generalize / reason about when 'model' is sometimes a model but sometimes a model parser.

For example, this case:
```
{
      "name": "gen_itinerary",
      "input": "Generate an itinerary ordered by {{order_by}} for these activities: {{get_activities.output}}.",
      "metadata": {
        "model": {
          "name": "AnyscaleEndpoint",
          "settings": {
            "model": "mistralai/Mixtral-8x7B-Instruct-v0.1"
          }
        },
        "parameters": {
          "order_by": "geographic location"
        }
      }
    }
```

Is not really necessary to support if the parser itself just handles the serialization of model name in a consistent way (always as the model name instead of in settings)

Reply:
I agree, I was thinking the same thing when I was working on this. Let me create an issue and we can work on improving this (and ideally having a single way)